include/RAJA/policy/atomic_builtin.hpp

```diff
@@ -219,17 +203,55 @@ template <typename T, typename OPER>
 RAJA_DEVICE_HIP RAJA_INLINE T builtin_atomic_CAS_oper(T volatile *acc,
                                                       OPER &&oper)
 {
-  BuiltinAtomicCAS<sizeof(T)> cas;
-  return cas(acc, std::forward<OPER>(oper), [](T const &) { return false; });
+  static_assert(sizeof(T) == 4 || sizeof(T) == 8,
+                "builtin atomic cas assumes 4 or 8 byte targets");
+
+#ifdef RAJA_COMPILER_MSVC
+#pragma warning( disable : 4244 )  // Force msvc to not emit conversion warning
+#endif
+  BuiltinAtomicType<T>* accConverted = (BuiltinAtomicType<T>*) acc;
+  BuiltinAtomicType<T> old = builtin_atomic_load(accConverted);
+  BuiltinAtomicType<T> expected;
+
+  do {
+    expected = old;
+    old = builtin_atomic_CAS(accConverted, expected, RAJA::util::reinterp_A_as_B<T, BuiltinAtomicType<T>>(oper(RAJA::util::reinterp_A_as_B<BuiltinAtomicType<T>, T>(expected))));
+  } while (old != expected);
+
+  return RAJA::util::reinterp_A_as_B<BuiltinAtomicType<T>, T>(old);
+#ifdef RAJA_COMPILER_MSVC
+#pragma warning( default : 4244 )  // Reenable warning
```

Comment:
😱

Reply:
I'm not sure if this warning is still an issue. Will the windows CI build show build warnings?

Reply:
I think it should in the raw log file. We should probably take a look at the sort of thing at some regular interval to squash `em if we can.

Reply:
Just removed the pragmas that disable the warnings. I'll check the logs to see if any warnings show up.

Reply:
It looks like this code path is not compiled in the windows CI builds. Would you rather I leave them out or put them back in?

Reply:
Is that because we are not testing something completely?

Reply:
Neither OpenMP nor the built in atomics are being tested on windows. Only the sequential policy is tested there.

Reply:
Should we add those items to the testing? The code project you work on is the main consumer of RAJA windows support. Does that project use those features?

Reply:
Currently, we're only using the sequential policies on Windows. I don't see us using the OpenMP policies any time soon. We are working on a CUDA build on Windows, though. It looks like the cuda atomic policies file includes the openmp and the builtin atomics files, but at first glance doesn't appear to use any of the builtin or openmp atomics. If it does use them, then we definitely want to add them to the testing.